types/controlled-frame.d.ts

```diff
@@ -14,170 +14,499 @@
-  canGoForward(): boolean;
-  forward(): Promise<void>;
-  go(relativeIndex?: number): Promise<void>;
-  reload(): void;
-  stop(): void;
+interface DialogEvent extends Event {
+  readonly messageType: DialogType;
+  readonly messageText: string;
+  readonly dialog: DialogController;
+}
 
-  // Scripting Methods
-  addContentScripts(contentScriptList: ContentScriptDetails[]): Promise<void>;
-  executeScript(details?: InjectDetails | {}): Promise<any>;
-  insertCSS(details?: InjectDetails): Promise<void>;
-  removeContentScripts(scriptNameList: string[]): Promise<void>;
+type WindowOpenDisposition =
+  | 'ignore'
+  | 'save_to_disk'
+  | 'current_tab'
+  | 'new_background_tab'
+  | 'new_foreground_tab'
+  | 'new_window'
+  | 'new_popup';
 
-  // Configuration methods
-  clearData(options: ClearDataOptions = {}): Promise<void>;
-  getAudioState(): Promise<boolean>;
-  getZoom(): Promise<number>;
-  isAudioMuted(): Promise;
-  setAudioMuted(mute: boolean): void;
-  setZoom(zoomFactor: number): Promise<void>;
+interface NewWindowController {
+  attach(newControlledFrame: HTMLControlledFrameElement): void;
+  discard(): void;
+}
 
-  // Capture methods
-  captureVisibleRegion(): void;
-  print(): void;
+interface NewWindowEvent extends Event {
+  readonly window: NewWindowController;
+  readonly targetUrl: string;
+  readonly name: string;
+  readonly windowOpenDisposition: WindowOpenDisposition;
+  readonly partition: string;
+  readonly initialHeight: number;
+  readonly initialWidth: number;
+}
+
+type PermissionType =
+  | 'media'
+  | 'geolocation'
+  | 'pointerLock'
+  | 'download'
+  | 'filesystem'
+  | 'fullscreen'
+  | 'hid';
+
+interface PermissionRequestController {
+  allow(): void;
+  cancel(): void;
+
+  readonly url: string;
+  readonly origin?: string;
+  readonly requestMethod?: string;
+  readonly lastUnlockedBySelf?: boolean;
+  readonly userGesture?: boolean;
+}
+
+interface PermissionRequestEvent extends Event {
+  readonly permission: PermissionType;
+  readonly request: PermissionRequestController;
+}
+
+interface SizeChangedEvent extends Event {
+  readonly oldWidth: number;
+  readonly oldHeight: number;
+  readonly newWidth: number;
+  readonly newHeight: number;
+}
+
+interface ZoomChangeEvent extends Event {
+  readonly oldZoomFactor: number;
+  readonly newZoomFactor: number;
+}
+
+interface ContentLoadEvent extends Event {}
+
+interface LoadAbortEvent extends Event {
+  readonly url: string;
+  readonly isTopLevel: boolean;
+  readonly code: number;
+  readonly reason: string;
+}
+
+interface LoadCommitEvent extends Event {
+  readonly url: string;
+  readonly isTopLevel: boolean;
+}
+
+interface LoadStartEvent extends Event {
+  readonly url: string;
+  readonly isTopLevel: boolean;
+}
+
+interface LoadStopEvent extends Event {}
+
+interface LoadRedirectEvent extends Event {
+  readonly oldUrl: string;
+  readonly newUrl: string;
+  readonly isTopLevel: boolean;
+}
+
+type ResourceType =
+  | 'main-frame'
```

Comment:
here it seams like we have the third type of enum values (`-`, `_`', camel case).